NAudio/Midi/MidiEvent.cs

```diff
@@ -192,6 +192,13 @@ public MidiEvent(long absoluteTime, int channel, MidiCommandCode commandCode)
         }
 
         /// <summary>
+        /// Creates a deep clone of this MIDI event.
+        /// </summary>
+        public virtual MidiEvent Clone() => (MidiEvent)MemberwiseClone();

```

Comment:
for now, let's get this PR in, and making MidiEvent abstract can be covered by future PR in which this method can be made abstract. Good news is that it is unlikely more derived MidiEvents will actually be created.